mesonbuild/backend/backends.py

```diff
@@ -1459,20 +1459,14 @@ def get_custom_target_sources(self, target: build.CustomTarget) -> T.List[str]:
             srcs += fname
         return srcs
 
-    def get_custom_target_depend_files(self, target: build.CustomTarget, absolute_paths: bool = False) -> T.List[str]:
+    def get_custom_target_depend_files(self, target: T.Union[build.CustomTarget, build.GeneratedList], absolute_paths: bool = False) -> T.List[str]:
         deps: T.List[str] = []
         for i in target.depend_files:
-            if isinstance(i, mesonlib.File):
```

Comment:
Deleting checks for depend_files that are strings, is a typing fix?

Reply:
depend_files is marked as a List of strings, but that is apparently wrong

Reply:
Note that issues like this are why I don't trust logic changes to be inside of type annotation fixes unless the commit message explains why that individual logic change is correct and the previous version was wrong. Type annotations are there to hint at what is possible, but they are not authoritative (at least not for a codebase that isn't fully typed yet).

Reply:
Some day...